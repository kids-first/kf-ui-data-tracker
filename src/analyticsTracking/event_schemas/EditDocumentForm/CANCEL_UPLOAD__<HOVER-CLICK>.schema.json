{
  "$id": "https://kf-ui-datatracker.kidsfirstdrc.org/event_schemas/CANCEL_UPLOAD/CANCEL_UPLOAD__<HOVER-CLICK>.schema.json",
  "$schema": "http://json-schema.org/draft-04/schema#",
  "title": "CANCEL_UPLOAD__<HOVER-CLICK>",
  "type": "object",
  "description": "onMouseOver event for the 'Cancel' button in the EditDocumentForm component of the NewDocumentView",
  "allOf": [
    {"$ref": "../common_defintions.json#/path"},
    {"$ref": "../common_defintions.json#/route"},
    {"$ref": "../common_defintions.json#/view"},
    {"$ref": "../BUTTON/button_defintions.json#/buton_text"},
    {"$ref": "../BUTTON/button_defintions.json#/buton_type"},
    {"$ref": "../BUTTON/button_defintions.json#/link"}
  ]
}
