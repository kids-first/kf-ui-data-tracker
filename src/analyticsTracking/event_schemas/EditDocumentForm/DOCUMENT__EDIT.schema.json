{
  "$id": "https://kf-ui-datatracker.kidsfirstdrc.org/event_schemas/DOCUMENT/DOCUMENT__EDIT.schema.json",
  "$schema": "http://json-schema.org/draft-04/schema#",
  "title": "DOCUMENT__EDIT",
  "type": "object",
  "description": "",
  "allOf": [
    {"$ref": "../common_defintions.json#/path"},
    {"$ref": "../common_defintions.json#/route"},
    {"$ref": "../common_defintions.json#/view"},
    {
      "properties": {
        "updated_version": {
          "type": "object",
          "properties": {
            "kfId": {
              "type": "string",
              "description": "kids first study file version identifier",
              "pattern": "^SF_[A-HJ-KM-NP-TV-Z0-9]{8}"
            },
            "file_name": {
              "type": "string",
              "description": "name of the local file associated with the file version"
            },
            "description": {
              "type": "string",
              "description": "user provided study document description field content"
            },
            "file_status": {
              "type": "string",
              "enum": ["PEN", "APP", "CHN", "PRC", "UPD"],
              "description": "Approval Status code"
            }
          }
        }
      }
    }
  ]
}
