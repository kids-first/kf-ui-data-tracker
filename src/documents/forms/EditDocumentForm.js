/* eslint-disable  no-useless-escape */
import React, {useState} from 'react';
import PropTypes from 'prop-types';
import {Form, Dropdown, Segment} from 'semantic-ui-react';
import {Formik, Field} from 'formik';
import validate from './validate';
import ExistingDocsMessage from './ExistingDocsMessage';
import SelectElement from '../components/FileDetail/SelectElement';
import Badge from '../../components/Badge/Badge';
import UploadWizard from '../modals/UploadWizard/UploadWizard';
import {fileTypeDetail, versionState} from '../../common/enums';
import MarkdownEditor from '../components/FileDetail/MarkdownEditor';
import draftToMarkdown from 'draftjs-to-markdown';
import {EditorState, convertToRaw, ContentState} from 'draft-js';
/**
 * Form to edit document information or annotate new file for uploading
 */
const EditDocumentForm = React.forwardRef(
  (
    {
      fileNode,
      fileType,
      fileName,
      fileDescription,
      versionStatus,
      history,
      handleSubmit,
      submitButtons,
      studyFiles,
    },
    ref,
  ) => {
    const [onUploading, setUploading] = useState(false);
    const [showDialog, setShowDialog] = useState(false);
    const [titleError, setTitleError] = useState({});
    const [editorState, setEditorState] = useState(
      EditorState.createWithContent(
        ContentState.createFromText(fileDescription || ''),
      ),
    );
    const options = ['PEN', 'APP', 'CHN', 'PRC'].map(state => ({
      key: state,
      value: state,
      text: versionState[state].title,
      content: <Badge state={state} />,
    }));
    return (
      <>
        <Formik
          initialValues={{
            file_name: fileName || '',
            file_type: fileType,
            file_desc: fileDescription,
            file_status: versionStatus,
          }}
          validate={vals => {
            setTitleError(validate(vals, fileNode, studyFiles));
            let errors = {};
            ['file_name', 'file_type'].forEach(function(field) {
              if (!vals[field]) {
                errors[field] = 'Required';
              }
            });
            const rawState = convertToRaw(editorState.getCurrentContent());
            const mdText = draftToMarkdown(rawState);
            if (mdText.length < 3) {
              errors.file_desc = 'Required';
            }
            return errors;
          }}
          onSubmit={values => {
            setUploading(true);
            const rawState = convertToRaw(editorState.getCurrentContent());
            const mdText = draftToMarkdown(rawState);
            var submitValue = values;
            submitValue.file_desc = mdText;
            handleSubmit(...Object.values(values));
          }}
        >
          {({
            values,
            handleChange,
            handleBlur,
            handleSubmit,
            setFieldValue,
            errors,
            touched,
            validateForm,
          }) => (
            <>
              {titleError.file_name &&
                (titleError.file_name.existing_similarity ||
                  titleError.file_name.exact_matches ||
                  titleError.file_name.upload_similarity) && (
                  <ExistingDocsMessage
                    setShowDialog={setShowDialog}
                    errors={titleError}
                    existingDocs={studyFiles}
                    fileNameInput={values.file_name}
                    newFile={!versionStatus}
                  />
                )}
              <Form onSubmit={handleSubmit} ref={ref}>
                <Form.Field required>
                  <label htmlFor="file_name">Document Title:</label>
                  <input
                    className={
                      touched.file_name && errors.file_name
                        ? 'border-red'
                        : null
                    }
                    data-testid="name-input"
                    type="text"
                    name="file_name"
                    placeholder="Phenotypic Data manifest for..."
                    value={values.file_name}
                    onBlur={handleBlur}
                    onChange={handleChange}
                  />
                  <small>
                    Please provide a descriptive title without dates or
                    adjectives such as "new", "updated", "final", etc.
                  </small>
                </Form.Field>
                {versionStatus && (
                  <Form.Field>
                    <label>Approval Status:</label>
                    <Dropdown
                      selection
                      fluid
                      name="file_status"
                      options={options}
                      value={values.file_status}
                      placeholder="Choose an option"
                      onChange={(e, {value}) => {
                        setFieldValue('file_status', value);
                      }}
                    />
                  </Form.Field>
                )}
                <Form.Field required>
                  <label htmlFor="file_type">Document Type:</label>
                  {Object.keys(fileTypeDetail).map(item => (
                    <Form.Field key={item}>
                      <Field
                        component={SelectElement}
                        name="file_type"
                        id={item}
                        label={item}
                      />
                    </Form.Field>
                  ))}
                </Form.Field>
                {submitButtons && (
                  <Form.Field required>
                    <label>Describe Document Contents:</label>
                    <Segment
                      className={
                        touched.file_desc && errors.file_desc
                          ? 'border-red'
                          : null
                      }
                    >
                      <MarkdownEditor
                        editorState={editorState}
                        onEditorStateChange={e => {
                          setEditorState(e);
                          validateForm();
                        }}
                      />
                    </Segment>
                  </Form.Field>
                )}
                {submitButtons &&
                  submitButtons(
                    Object.keys(errors).length > 0 ||
                      values.file_name.length === 0,
                    onUploading,
                  )}
              </Form>
            </>
          )}
        </Formik>
        {showDialog && (
          <UploadWizard
            startingStep={1}
            studyId={history.location.pathname.split('/')[2]}
            onCloseDialog={() => {
              setShowDialog(false);
            }}
            file={fileNode}
            fileList={studyFiles}
            history={history}
          />
        )}
      </>
    );
  },
);

EditDocumentForm.propTypes = {
  /** The title of the file */
  fileName: PropTypes.string,
  /** The description for the file */
  fileDescription: PropTypes.string,
  /** The value of the currently selected fileType's enum */
  fileType: PropTypes.string,
  /** The file approval status */
  versionStatus: PropTypes.string,
  /** If the user has ADMIN role */
  handleSubmit: PropTypes.func,
  /** (New file) Any errors that occured when submitting the form */
  errors: PropTypes.object,
  /** (New file) Displays as buttons for form submitting */
  submitButtons: PropTypes.func,
};

export default EditDocumentForm;
