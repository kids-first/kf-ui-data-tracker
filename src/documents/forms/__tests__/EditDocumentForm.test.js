import React from 'react';
import {render} from '@testing-library/react';
import EditDocumentForm from '../EditDocumentForm';

it('renders correctly for annotating new file before upload', () => {
  const cancelMock = jest.fn();
  const submitMock = jest.fn();
  const tree = render(
    <EditDocumentForm handleCancel={cancelMock} handleSubmit={submitMock} />,
  );
  expect(tree).toMatchSnapshot();
});

it('renders correctly for editing existing file', () => {
  const tree = render(
    <EditDocumentForm
      fileType="CLN"
      fileName="Existing File Name"
      fileDescription="Existing File Description"
      versionStatus="PEN"
      onNameChange={e => e.preventDefault()}
      onDescriptionChange={e => e.preventDefault()}
      onFileTypeChange={e => e.preventDefault()}
      onVersionStatusChange={versionStatusValue => {}}
    />,
  );
  expect(tree).toMatchSnapshot();
});
