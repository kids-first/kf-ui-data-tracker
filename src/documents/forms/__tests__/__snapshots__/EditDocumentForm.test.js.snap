// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for annotating new file before upload 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="ui form"
      >
        <div
          class="required field"
        >
          <label
            for="file_name"
          >
            Document Title:
          </label>
          <input
            data-testid="name-input"
            error="undefined"
            name="file_name"
            placeholder="Phenotypic Data manifest for..."
            type="text"
            value=""
          />
        </div>
        <div
          class="required field"
        >
          <label
            for="file_type"
          >
            Document Type:
          </label>
          <div
            class="field"
          >
            <div
              class="ui compact segment selectionRadio--card"
            >
              <div>
                <div
                  class="ui radio checkbox"
                >
                  <input
                    class="hidden"
                    id="SHM"
                    name="file_type"
                    readonly=""
                    tabindex="0"
                    type="radio"
                    value="SHM"
                  />
                  <i
                    aria-hidden="true"
                    class="black shipping large bordered circular inverted icon"
                  />
                </div>
              </div>
              <div>
                <p
                  class="mb-5"
                >
                  <b>
                    Biospecimen Manifest
                  </b>
                </p>
                <p
                  class="text-wrap-75 text-grey"
                >
                  Tabular files (e.g. CSV, TSV, Excel) containing biospecimen identifiers and metadata (e.g. tissue type, composition) provided to sequencing centers as well as consent type.
                </p>
              </div>
            </div>
          </div>
          <div
            class="field"
          >
            <div
              class="ui compact segment selectionRadio--card"
            >
              <div>
                <div
                  class="ui radio checkbox"
                >
                  <input
                    class="hidden"
                    id="CLN"
                    name="file_type"
                    readonly=""
                    tabindex="0"
                    type="radio"
                    value="CLN"
                  />
                  <i
                    aria-hidden="true"
                    class="black hospital large bordered circular inverted icon"
                  />
                </div>
              </div>
              <div>
                <p
                  class="mb-5"
                >
                  <b>
                    Clinical/Phenotypic Data
                  </b>
                </p>
                <p
                  class="text-wrap-75 text-grey"
                >
                  Structured files with data linked to the biospecimen identifiers provided in the shipping manifest(s). Exports from data management applications, such as REDCap, are the fastest to process, but we will work with you on most formats.
                </p>
              </div>
            </div>
          </div>
          <div
            class="field"
          >
            <div
              class="ui compact segment selectionRadio--card"
            >
              <div>
                <div
                  class="ui radio checkbox"
                >
                  <input
                    class="hidden"
                    id="SEQ"
                    name="file_type"
                    readonly=""
                    tabindex="0"
                    type="radio"
                    value="SEQ"
                  />
                  <i
                    aria-hidden="true"
                    class="black dna large bordered circular inverted icon"
                  />
                </div>
              </div>
              <div>
                <p
                  class="mb-5"
                >
                  <b>
                    Sequencing Manifest
                  </b>
                </p>
                <p
                  class="text-wrap-75 text-grey"
                >
                  Tabular files (e.g. CSV, TSV, Excel) containing the biospecimen identifiers and the associated genomic data files. Typically provided by the sequencing center.
                </p>
              </div>
            </div>
          </div>
          <div
            class="field"
          >
            <div
              class="ui compact segment selectionRadio--card"
            >
              <div>
                <div
                  class="ui radio checkbox"
                >
                  <input
                    class="hidden"
                    id="OTH"
                    name="file_type"
                    readonly=""
                    tabindex="0"
                    type="radio"
                    value="OTH"
                  />
                  <i
                    aria-hidden="true"
                    class="black question large bordered circular inverted icon"
                  />
                </div>
              </div>
              <div>
                <p
                  class="mb-5"
                >
                  <b>
                    Other
                  </b>
                </p>
                <p
                  class="text-wrap-75 text-grey"
                >
                  Any useful documents, such as study background information, data dictionaries, etc., that does not clearly fit in the other categories.
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="required field"
        >
          <label>
            Describe Document Contents:
          </label>
          <textarea
            data-testid="description-input"
            name="file_desc"
            rows="3"
            type="text"
          />
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="ui form"
    >
      <div
        class="required field"
      >
        <label
          for="file_name"
        >
          Document Title:
        </label>
        <input
          data-testid="name-input"
          error="undefined"
          name="file_name"
          placeholder="Phenotypic Data manifest for..."
          type="text"
          value=""
        />
      </div>
      <div
        class="required field"
      >
        <label
          for="file_type"
        >
          Document Type:
        </label>
        <div
          class="field"
        >
          <div
            class="ui compact segment selectionRadio--card"
          >
            <div>
              <div
                class="ui radio checkbox"
              >
                <input
                  class="hidden"
                  id="SHM"
                  name="file_type"
                  readonly=""
                  tabindex="0"
                  type="radio"
                  value="SHM"
                />
                <i
                  aria-hidden="true"
                  class="black shipping large bordered circular inverted icon"
                />
              </div>
            </div>
            <div>
              <p
                class="mb-5"
              >
                <b>
                  Biospecimen Manifest
                </b>
              </p>
              <p
                class="text-wrap-75 text-grey"
              >
                Tabular files (e.g. CSV, TSV, Excel) containing biospecimen identifiers and metadata (e.g. tissue type, composition) provided to sequencing centers as well as consent type.
              </p>
            </div>
          </div>
        </div>
        <div
          class="field"
        >
          <div
            class="ui compact segment selectionRadio--card"
          >
            <div>
              <div
                class="ui radio checkbox"
              >
                <input
                  class="hidden"
                  id="CLN"
                  name="file_type"
                  readonly=""
                  tabindex="0"
                  type="radio"
                  value="CLN"
                />
                <i
                  aria-hidden="true"
                  class="black hospital large bordered circular inverted icon"
                />
              </div>
            </div>
            <div>
              <p
                class="mb-5"
              >
                <b>
                  Clinical/Phenotypic Data
                </b>
              </p>
              <p
                class="text-wrap-75 text-grey"
              >
                Structured files with data linked to the biospecimen identifiers provided in the shipping manifest(s). Exports from data management applications, such as REDCap, are the fastest to process, but we will work with you on most formats.
              </p>
            </div>
          </div>
        </div>
        <div
          class="field"
        >
          <div
            class="ui compact segment selectionRadio--card"
          >
            <div>
              <div
                class="ui radio checkbox"
              >
                <input
                  class="hidden"
                  id="SEQ"
                  name="file_type"
                  readonly=""
                  tabindex="0"
                  type="radio"
                  value="SEQ"
                />
                <i
                  aria-hidden="true"
                  class="black dna large bordered circular inverted icon"
                />
              </div>
            </div>
            <div>
              <p
                class="mb-5"
              >
                <b>
                  Sequencing Manifest
                </b>
              </p>
              <p
                class="text-wrap-75 text-grey"
              >
                Tabular files (e.g. CSV, TSV, Excel) containing the biospecimen identifiers and the associated genomic data files. Typically provided by the sequencing center.
              </p>
            </div>
          </div>
        </div>
        <div
          class="field"
        >
          <div
            class="ui compact segment selectionRadio--card"
          >
            <div>
              <div
                class="ui radio checkbox"
              >
                <input
                  class="hidden"
                  id="OTH"
                  name="file_type"
                  readonly=""
                  tabindex="0"
                  type="radio"
                  value="OTH"
                />
                <i
                  aria-hidden="true"
                  class="black question large bordered circular inverted icon"
                />
              </div>
            </div>
            <div>
              <p
                class="mb-5"
              >
                <b>
                  Other
                </b>
              </p>
              <p
                class="text-wrap-75 text-grey"
              >
                Any useful documents, such as study background information, data dictionaries, etc., that does not clearly fit in the other categories.
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="required field"
      >
        <label>
          Describe Document Contents:
        </label>
        <textarea
          data-testid="description-input"
          name="file_desc"
          rows="3"
          type="text"
        />
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders correctly for editing existing file 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="ui form"
      >
        <div
          class="required field"
        >
          <label
            for="file_name"
          >
            Document Title:
          </label>
          <input
            data-testid="name-input"
            error="undefined"
            name="file_name"
            placeholder="Phenotypic Data manifest for..."
            type="text"
            value=""
          />
        </div>
        <div
          class="required field"
        >
          <label
            for="file_type"
          >
            Document Type:
          </label>
          <div
            class="field"
          >
            <div
              class="ui compact segment selectionRadio--card"
            >
              <div>
                <div
                  class="ui radio checkbox"
                >
                  <input
                    class="hidden"
                    id="SHM"
                    name="file_type"
                    readonly=""
                    tabindex="0"
                    type="radio"
                    value="SHM"
                  />
                  <i
                    aria-hidden="true"
                    class="black shipping large bordered circular inverted icon"
                  />
                </div>
              </div>
              <div>
                <p
                  class="mb-5"
                >
                  <b>
                    Biospecimen Manifest
                  </b>
                </p>
                <p
                  class="text-wrap-75 text-grey"
                >
                  Tabular files (e.g. CSV, TSV, Excel) containing biospecimen identifiers and metadata (e.g. tissue type, composition) provided to sequencing centers as well as consent type.
                </p>
              </div>
            </div>
          </div>
          <div
            class="field"
          >
            <div
              class="ui compact segment selectionRadio--card"
            >
              <div>
                <div
                  class="ui radio checkbox"
                >
                  <input
                    class="hidden"
                    id="CLN"
                    name="file_type"
                    readonly=""
                    tabindex="0"
                    type="radio"
                    value="CLN"
                  />
                  <i
                    aria-hidden="true"
                    class="black hospital large bordered circular inverted icon"
                  />
                </div>
              </div>
              <div>
                <p
                  class="mb-5"
                >
                  <b>
                    Clinical/Phenotypic Data
                  </b>
                </p>
                <p
                  class="text-wrap-75 text-grey"
                >
                  Structured files with data linked to the biospecimen identifiers provided in the shipping manifest(s). Exports from data management applications, such as REDCap, are the fastest to process, but we will work with you on most formats.
                </p>
              </div>
            </div>
          </div>
          <div
            class="field"
          >
            <div
              class="ui compact segment selectionRadio--card"
            >
              <div>
                <div
                  class="ui radio checkbox"
                >
                  <input
                    class="hidden"
                    id="SEQ"
                    name="file_type"
                    readonly=""
                    tabindex="0"
                    type="radio"
                    value="SEQ"
                  />
                  <i
                    aria-hidden="true"
                    class="black dna large bordered circular inverted icon"
                  />
                </div>
              </div>
              <div>
                <p
                  class="mb-5"
                >
                  <b>
                    Sequencing Manifest
                  </b>
                </p>
                <p
                  class="text-wrap-75 text-grey"
                >
                  Tabular files (e.g. CSV, TSV, Excel) containing the biospecimen identifiers and the associated genomic data files. Typically provided by the sequencing center.
                </p>
              </div>
            </div>
          </div>
          <div
            class="field"
          >
            <div
              class="ui compact segment selectionRadio--card"
            >
              <div>
                <div
                  class="ui radio checkbox"
                >
                  <input
                    class="hidden"
                    id="OTH"
                    name="file_type"
                    readonly=""
                    tabindex="0"
                    type="radio"
                    value="OTH"
                  />
                  <i
                    aria-hidden="true"
                    class="black question large bordered circular inverted icon"
                  />
                </div>
              </div>
              <div>
                <p
                  class="mb-5"
                >
                  <b>
                    Other
                  </b>
                </p>
                <p
                  class="text-wrap-75 text-grey"
                >
                  Any useful documents, such as study background information, data dictionaries, etc., that does not clearly fit in the other categories.
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="required field"
        >
          <label>
            Describe Document Contents:
          </label>
          <textarea
            data-testid="description-input"
            name="file_desc"
            rows="3"
            type="text"
          />
        </div>
      </form>
    </div>
    <div>
      <form
        class="ui form"
      >
        <div
          class="required field"
        >
          <label
            for="file_name"
          >
            Document Title:
          </label>
          <input
            data-testid="name-input"
            error="undefined"
            name="file_name"
            placeholder="Phenotypic Data manifest for..."
            type="text"
            value="Existing File Name"
          />
        </div>
        <div
          class="field"
        >
          <label>
            Approval Status:
          </label>
          <div
            class="ui orange tiny basic label"
          >
            Pending review
          </div>
        </div>
        <div
          class="required field"
        >
          <label
            for="file_type"
          >
            Document Type:
          </label>
          <div
            class="field"
          >
            <div
              class="ui compact segment selectionRadio--card"
            >
              <div>
                <div
                  class="ui radio checkbox"
                >
                  <input
                    class="hidden"
                    id="SHM"
                    name="file_type"
                    readonly=""
                    tabindex="0"
                    type="radio"
                    value="SHM"
                  />
                  <i
                    aria-hidden="true"
                    class="black shipping large bordered circular inverted icon"
                  />
                </div>
              </div>
              <div>
                <p
                  class="mb-5"
                >
                  <b>
                    Biospecimen Manifest
                  </b>
                </p>
                <p
                  class="text-wrap-75 text-grey"
                >
                  Tabular files (e.g. CSV, TSV, Excel) containing biospecimen identifiers and metadata (e.g. tissue type, composition) provided to sequencing centers as well as consent type.
                </p>
              </div>
            </div>
          </div>
          <div
            class="field"
          >
            <div
              class="ui blue compact segment selectionRadio--card"
            >
              <div>
                <div
                  class="ui checked radio checkbox"
                >
                  <input
                    checked=""
                    class="hidden"
                    id="CLN"
                    name="file_type"
                    readonly=""
                    tabindex="0"
                    type="radio"
                    value="CLN"
                  />
                  <i
                    aria-hidden="true"
                    class="blue hospital large bordered circular inverted icon"
                  />
                </div>
              </div>
              <div>
                <p
                  class="mb-5"
                >
                  <b>
                    Clinical/Phenotypic Data
                  </b>
                </p>
                <p
                  class="text-wrap-75 text-grey"
                >
                  Structured files with data linked to the biospecimen identifiers provided in the shipping manifest(s). Exports from data management applications, such as REDCap, are the fastest to process, but we will work with you on most formats.
                </p>
              </div>
            </div>
          </div>
          <div
            class="field"
          >
            <div
              class="ui compact segment selectionRadio--card"
            >
              <div>
                <div
                  class="ui radio checkbox"
                >
                  <input
                    class="hidden"
                    id="SEQ"
                    name="file_type"
                    readonly=""
                    tabindex="0"
                    type="radio"
                    value="SEQ"
                  />
                  <i
                    aria-hidden="true"
                    class="black dna large bordered circular inverted icon"
                  />
                </div>
              </div>
              <div>
                <p
                  class="mb-5"
                >
                  <b>
                    Sequencing Manifest
                  </b>
                </p>
                <p
                  class="text-wrap-75 text-grey"
                >
                  Tabular files (e.g. CSV, TSV, Excel) containing the biospecimen identifiers and the associated genomic data files. Typically provided by the sequencing center.
                </p>
              </div>
            </div>
          </div>
          <div
            class="field"
          >
            <div
              class="ui compact segment selectionRadio--card"
            >
              <div>
                <div
                  class="ui radio checkbox"
                >
                  <input
                    class="hidden"
                    id="OTH"
                    name="file_type"
                    readonly=""
                    tabindex="0"
                    type="radio"
                    value="OTH"
                  />
                  <i
                    aria-hidden="true"
                    class="black question large bordered circular inverted icon"
                  />
                </div>
              </div>
              <div>
                <p
                  class="mb-5"
                >
                  <b>
                    Other
                  </b>
                </p>
                <p
                  class="text-wrap-75 text-grey"
                >
                  Any useful documents, such as study background information, data dictionaries, etc., that does not clearly fit in the other categories.
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="required field"
        >
          <label>
            Describe Document Contents:
          </label>
          <textarea
            data-testid="description-input"
            name="file_desc"
            rows="3"
            type="text"
          >
            Existing File Description
          </textarea>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="ui form"
    >
      <div
        class="required field"
      >
        <label
          for="file_name"
        >
          Document Title:
        </label>
        <input
          data-testid="name-input"
          error="undefined"
          name="file_name"
          placeholder="Phenotypic Data manifest for..."
          type="text"
          value="Existing File Name"
        />
      </div>
      <div
        class="field"
      >
        <label>
          Approval Status:
        </label>
        <div
          class="ui orange tiny basic label"
        >
          Pending review
        </div>
      </div>
      <div
        class="required field"
      >
        <label
          for="file_type"
        >
          Document Type:
        </label>
        <div
          class="field"
        >
          <div
            class="ui compact segment selectionRadio--card"
          >
            <div>
              <div
                class="ui radio checkbox"
              >
                <input
                  class="hidden"
                  id="SHM"
                  name="file_type"
                  readonly=""
                  tabindex="0"
                  type="radio"
                  value="SHM"
                />
                <i
                  aria-hidden="true"
                  class="black shipping large bordered circular inverted icon"
                />
              </div>
            </div>
            <div>
              <p
                class="mb-5"
              >
                <b>
                  Biospecimen Manifest
                </b>
              </p>
              <p
                class="text-wrap-75 text-grey"
              >
                Tabular files (e.g. CSV, TSV, Excel) containing biospecimen identifiers and metadata (e.g. tissue type, composition) provided to sequencing centers as well as consent type.
              </p>
            </div>
          </div>
        </div>
        <div
          class="field"
        >
          <div
            class="ui blue compact segment selectionRadio--card"
          >
            <div>
              <div
                class="ui checked radio checkbox"
              >
                <input
                  checked=""
                  class="hidden"
                  id="CLN"
                  name="file_type"
                  readonly=""
                  tabindex="0"
                  type="radio"
                  value="CLN"
                />
                <i
                  aria-hidden="true"
                  class="blue hospital large bordered circular inverted icon"
                />
              </div>
            </div>
            <div>
              <p
                class="mb-5"
              >
                <b>
                  Clinical/Phenotypic Data
                </b>
              </p>
              <p
                class="text-wrap-75 text-grey"
              >
                Structured files with data linked to the biospecimen identifiers provided in the shipping manifest(s). Exports from data management applications, such as REDCap, are the fastest to process, but we will work with you on most formats.
              </p>
            </div>
          </div>
        </div>
        <div
          class="field"
        >
          <div
            class="ui compact segment selectionRadio--card"
          >
            <div>
              <div
                class="ui radio checkbox"
              >
                <input
                  class="hidden"
                  id="SEQ"
                  name="file_type"
                  readonly=""
                  tabindex="0"
                  type="radio"
                  value="SEQ"
                />
                <i
                  aria-hidden="true"
                  class="black dna large bordered circular inverted icon"
                />
              </div>
            </div>
            <div>
              <p
                class="mb-5"
              >
                <b>
                  Sequencing Manifest
                </b>
              </p>
              <p
                class="text-wrap-75 text-grey"
              >
                Tabular files (e.g. CSV, TSV, Excel) containing the biospecimen identifiers and the associated genomic data files. Typically provided by the sequencing center.
              </p>
            </div>
          </div>
        </div>
        <div
          class="field"
        >
          <div
            class="ui compact segment selectionRadio--card"
          >
            <div>
              <div
                class="ui radio checkbox"
              >
                <input
                  class="hidden"
                  id="OTH"
                  name="file_type"
                  readonly=""
                  tabindex="0"
                  type="radio"
                  value="OTH"
                />
                <i
                  aria-hidden="true"
                  class="black question large bordered circular inverted icon"
                />
              </div>
            </div>
            <div>
              <p
                class="mb-5"
              >
                <b>
                  Other
                </b>
              </p>
              <p
                class="text-wrap-75 text-grey"
              >
                Any useful documents, such as study background information, data dictionaries, etc., that does not clearly fit in the other categories.
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="required field"
      >
        <label>
          Describe Document Contents:
        </label>
        <textarea
          data-testid="description-input"
          name="file_desc"
          rows="3"
          type="text"
        >
          Existing File Description
        </textarea>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
